test/verify/check-examples

```diff
@@ -101,7 +101,7 @@ class TestLongRunning(MachineCase):
         m.execute("mkdir -p ~admin/.local/share/cockpit")
         m.upload([os.path.join(EXAMPLES_DIR, "long-running-process")], "/home/admin/.local/share/cockpit/")
         # clean up after test failures
-        self.addCleanup(m.execute, "systemctl stop cockpit-longrunning.service 2>/dev/null && systemctl reset-failed cockpit-longrunning.service || true")
+        self.addCleanup(m.execute, "systemctl stop cockpit-longrunning.service 2>/dev/null || true; systemctl reset-failed cockpit-longrunning.service || true")
```

Comment:
I feel like this is a good example of something that's not really improved by this change.

Splitting it into two separate `addCleanup()` would be my first suggestion, but then I don't know if that function is run-first-first or run-last-first, and the fact that I'd need to check for it is already an argument against doing that.

I almost find myself wanting a version of m.execute() that accepts an array — but instead of being an argument vector, it would be an array of commands to run.  But that's gross too.

Why not make better use of `check=False` for the cases we don't care about failures for?  For that matter, what if we caused `check=False` to have the side effect of not running `set -e`?

Reply:
I disagree about "not an improvement" -- by now I have found several instances where the second command was silently skipped in case of the first one failing. It's not fully clear whether it would break this particular case -- the stop command *definitively* fails in many cases (that was one of the force-pushes). It was never the intention to skip the reset-failed in that case.

I mostly care about having these opportunistic cleanups be simply greppable, `|| true` in this case. A `check=False` would work as well, it's just less flexible in the case of multiple commands. Not my prefered solution (unnecessary API complexity), but I don't have a strong opinion about it, i.e. no veto from me.

Reply:
I looked at this again with fresh eyes, considered alternatives like `check=True` or prepending `set +e`, and I still prefer `|| true` as the most explicit and most robust variant. So I just resolved the conflict for now.

Reply:
I dislike `check=True` in particular because it's much less obvious: Does it ignore non-zero exit codes of the whole Popen command (which would be my first instinct on a Python API)? That's not what it does, it's equivalent to `set +e` -- but that isn't reflected in the name. I really like the simplicity and clarity of `.execute()` being "a strict `set -e` script, period". `check=False` is harder to grep and visually scan for (as it often is on a separate line), and not self-descriptive enough IMHO.